lib/bamboo/adapters/smtp_adapter.ex

```diff
@@ -161,6 +168,45 @@ defmodule Bamboo.SMTPAdapter do
     |> add_smtp_line(text_body)
   end
 
+  defp add_multipart_mixed_split(body, delimiter), do: "#{body}\r\n--#{delimiter}\r\n"
+
+  defp add_attachment_header(body, attachment) do
+    << random :: size(32) >> = :crypto.strong_rand_bytes(4)
+    "#{body}Content-Type: attachment.content_type; name=\"#{attachment.filename}\"\r\n" <>
+    "Content-Disposition: attachment; filename=\"#{attachment.filename}\"\r\n" <>
+    "Content-Transfer-Encoding: base64\r\n" <>
+    "X-Attachment-Id: #{random}\r\n"
+  end
+
+  defp add_attachment_body(body, attachment) do
+    data = case attachment.path do
+      {:binary, data} -> data
+      _ -> File.read!(attachment.path)
+    end
+    "#{body}\r\n#{Base.encode64(data)}\r\n"
```

Comment:
1. `body` already ends with a newline, do we need to add one blank line between the body and the attachment?
     I guess it's useful in order to separate the header from the body. In this case I would suggest to add a blank line (i.e: `|> add_smtp_line("")`) in the `add_attachment` function (between the header and body function calls).
    It would make it obvious we want an empty line between header and body. Moreover, it would be more consistent with what we do elsewhere
2. What do you think of using `add_smtp_line` instead of doing the concatenation yourself?

```elixir
add_smtp_line(body, Base.encode64(data))
```